lib/rpc/server/agent.go

```diff
@@ -136,7 +137,15 @@ func (srv *agentServer) GetCurrentTime(ctx context.Context, _ *types.Empty) (*ty
 // Shutdown requests agent to shut down
 func (srv *agentServer) Shutdown(ctx context.Context, _ *types.Empty) (*types.Empty, error) {
 	srv.Info("Shutdown.")
-	go srv.Stop(ctx)
+	go func() {
+		// Create a separate context from the parent one since the parent
+		// context is canceled once the handler has returned
+		ctx, cancel := context.WithTimeout(context.Background(), defaults.ShutdownTimeout)
+		if err := srv.Stop(ctx); err != nil {
+			srv.Warnf("Failed to shutdown: %v.", err)
```

Comment:
srv.WithError(err).Warn()?

Reply:
This is not critical and will likely be spam.